packages/flutter_tools/lib/src/dart/language_version.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:package_config/package_config.dart';
+
+final RegExp _languageVersion = RegExp('/\/\\s*@dart');
```

Comment:
This regex looks like it's missing an initial escape `\` and has too many `\` in the middle, but it's obviously working?
`\/\/\s*@dart`

Also:
https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/feature-specification.md#individual-library-language-version-override
```
it would be possible to have a line containing just //@dart=2.3 inside a block comment. That will not count as a language version override marker.
```
Also need to check for block comments?  It would be unlikely someone would have a block comment containing it above the imports but hey that's the spec.

Reply:
@jmagman I went through so many iterations of the RegExp I considered lifting the actual parser from the SDK

Reply:
Anyway, will check for block comments too

Reply:
This still looks malformed with the unescaped initial `/`.  Check me @christopherfujino, shouldn't this be `\/\/\s*@dart`?  How is this working?

Reply:
Yeah, I looked at earlier, and agreed both that it looks wrong and that I don't know how it works. Lemme dig a little deeper...

Reply:
I agree it is completely bonkers. Confusingly the obvious regex pattern didn't work, and I haven't been able to figure out why

Reply:
@jonahwilliams ah! it's because you didn't use a raw string!

Reply:
(thank you vim dart syntax highlighting @natebosch)